pkg/staticnetworkconfig/generator.go

```diff
@@ -148,6 +145,27 @@ func (s *StaticNetworkConfigGenerator) formatNMConnection(nmConnection string) (
 	return buf.String(), nil
 }
 
+func (s *StaticNetworkConfigGenerator) validateAndCalculateMAC(ifaceName string) (string, error) {
+	// check the format of the interface name
+	match, err := regexp.MatchString("^[0-9A-Fa-f]{12}$", ifaceName)
+	if err != nil {
+		s.log.WithError(err).Errorf("Invalid regexp expression ^[0-9A-Fa-f]{12}$")
```

Comment:
I'd suggest to return only the error and let the caller to be responsible for printing the message

Reply:
i prefer to keep it this way, it will be more clear in the log which error occured